### DISCUSSION TITLE
Change per page record count

### DISCUSSION THREAD

**Discussion by jbk68** [cortexrd/Knack-Toolkit-Library#344] [General]
Looking for a way to increase that to 2500, 5000 or even better “Show All”

Didn't know if KTL currently has this or if this is something that can be added.

If anyone knows a way of adding it and has a script, it would greatly appreciated if you go share it with me.

Any and all help on this would be appreciated… JON

**Comment by CSWinnall** [comment]
Hi @jbk68 

As far as I know Knack has a limit of 1000 records. So the KTL will too.  The issue will be load time if it was possible.  If you want to load more records you may need to look into API and load more records as you scroll (infinite scroll or lazy load records).

Craig

**Reply by CSWinnall** [reply]
The other option is filtering records or using KTL public filters so uses can quickly zoom in on certain records.
Another option is search tables to refine the records.  

Why would you want to show that many records.  If you give more context there may be other ways to help.

**Comment by jbk68** [comment]
Thanks Craig for the responses.  The only reason I want to be able to view
or load more records than 1000 at a time is because I have created an
export button that exports my data to .xlxs format with selected table data
and due the fact I can only view 1000 records max, I can only export 1000
records max at a single time.  So when I have a table with say 5000
records, I literally have to export 5 times (page1, page2 etc...)

**Reply by CSWinnall** [reply]
Hi Jon

When you use _dr in the URL you should see where the number of rows is.  It's something like

rows_per_page =1000

Have you tried changing that to a higher number to see if it is possible.

Craig

**Comment by jbk68** [comment]
Hey Craig, thanks again for assistance.  I have not tried that.  Can you
please explain were to place that keyword?

**Comment by cortexrd** [comment]
Hi guys,

I tried several things, including deep exploration of Knack's code and real-time value changes to see if I could get a higher count of displayed records, but that seems to be limited internally at the server-side, no matter what you do.

As Craig pointed out, the only way to do this is to automate sequential multi-page fetches.  I would concatenate them in memory, then do the export.  I was planning to do this, but since Knack said they already have this on the development timetable, I decided to concentrate on other things.

Norm

**Comment by CSWinnall** [comment]
Hi Jon

You can place

_dr=1000

In the description of the view.

I've just seen Norms reply so it's not going to work.  

Another option would be to use make.com when you click the button it will send a webhook to make then you could do sequential searches for records then import them to a spreadsheet but this would be operation intensive.
